doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,183 @@
+order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+*note: each box will be referred to as a "phase" of the event loop.*
+
+*There is a slight discrepancy between the Windows and the Unix/Linux
+implementation, but that's not important for this demonstration.  The most
+important parts are here.  There are actually seven or eight steps, but the
+ones we care about — ones that Node.js actually uses are these four.*
+
+## timers
+
+This phase executes callbacks scheduled by `setTimeout()` and `setInterval()`.
+
+When you create a timer, you make a call to `setTimeout()`.  Then, when
+the poll phase of the event loop is entered, the number of ms before the 
+soonest timer is to be called is set as the poll's timeout. Meaning the 
+poll phase will return after "timeout" ms.  After that many 
+milliseconds, the `poll` phase will return and wrap back around to the 
+timers phase where those callbacks can be processed.
+
+Take note that the poll phase can only return while idle; execution of 
+a callback is allowed to run to completion, and can cause  unexpected 
+delay running the timer. 
+
+*Note: The `poll` phase technically controls when timers are called due to its
+ability to cause a thread to sit idly without burning CPU in order to stall the
+event loop so the timer can execute.*
+
+## pending callbacks:
+
+This phase executes callbacks for specific types of TCP errors, for example.
+
+## poll:
+
+This is the phase in which the event loop sits and waits for incoming
+connections to be received.  Ideally, most scripts spend most of their time
+here.
+
+## `setImmediate()`:
+
+`setImmediate()` is actually a special timer that runs in a separate phase of
+the event loop.  It uses a libuv API that schedules callbacks to execute after
+the poll phase has completed.
+
+Generally, as the code is executed, the event loop will eventually hit the
+`poll` phase where it will wait for an incoming connection, request, etc.
+However, after a callback has been scheduled with `setImmediate()`, at the
+start of the poll phase, a check will be run to see if there are any callbacks
+waiting.  If there are none waiting, the poll phase will end and continue to
+the `setImmediate` callback phase.
+
+### `setImmediate()` vs `setTimeout()`
+
+How quickly a `setImmediate()` callback is executed is only limited by how
+quickly the event loop can be processed whereas a timer won't fire until the
+number of milliseconds passed have elapsed.
```

Comment:
As a bit of background, it would be more technically correct to say 'at least the number of milliseconds have passed' because timers can fire late due to process/thread scheduling issues.